azure/services/resourceskus/sku.go

```diff
@@ -68,6 +68,10 @@ const (
 	MaximumPlatformFaultDomainCount = "MaximumPlatformFaultDomainCount"
 	// UltraSSDAvailable identifies the capability for the support of UltraSSD data disks.
 	UltraSSDAvailable = "UltraSSDAvailable"
+	// TrustedLaunchDisabled identifies the absence of the trusted launch capability.
+	TrustedLaunchDisabled = "TrustedLaunchDisabled"
+	// ConfidentialComputingType identifies the capability for confidentical computing.
+	ConfidentialComputingType = "ConfidentialComputingType"
```

Comment:
Jus curious about these values. `TrustedLaunchDisabled` and `ConfidentialComputingType`. Are these exposed by Azure SDK?

Reply:
Great question, I should have mentioned something in the PR description or via a code comment. I found these values by querying the [Compute](https://learn.microsoft.com/en-us/rest/api/compute/resource-skus/list) Azure REST API.

If I understand correctly from this [discussion](https://github.com/Azure/azure-rest-api-specs/issues/10033#issuecomment-657701452), the resource SKUs list is dynamic because it's quite hard to maintain a hardcoded enum. So we don't/won't see them declared either in the Azure Go SDK or the Azure REST API specs.

Nevertheless I thought it's a useful validation to have in order to fail early enough. What do you think?

Reply:
I am having trouble capturing the response of 
`GET https://management.azure.com/subscriptions/{subscriptionId}/providers/Microsoft.Compute/skus?api-version=2021-07-01`  as mentioned on [Compute reference docs](https://learn.microsoft.com/en-us/rest/api/compute/resource-skus/list). If you don't mind, can you please paste a response you see? I am interested in seeing `ConfidentialComputingType` in the response. 

However, I agree that these are useful validations.

Reply:
Of course, below you can find a single item with an `SNP` `ConfidentialComputingType` capability, by running `az vm list-skus --location eastus`:

```json
[
  {
    "apiVersions": null,
    "capabilities": [
      {
        "name": "MaxResourceVolumeMB",
        "value": "768000"
      },
      {
        "name": "OSVhdSizeMB",
        "value": "1047552"
      },
      {
        "name": "vCPUs",
        "value": "20"
      },
      {
        "name": "MemoryPreservingMaintenanceSupported",
        "value": "True"
      },
      {
        "name": "HyperVGenerations",
        "value": "V1,V2"
      },
      {
        "name": "MemoryGB",
        "value": "160"
      },
      {
        "name": "MaxDataDiskCount",
        "value": "32"
      },
      {
        "name": "CpuArchitectureType",
        "value": "x64"
      },
      {
        "name": "LowPriorityCapable",
        "value": "True"
      },
      {
        "name": "PremiumIO",
        "value": "True"
      },
      {
        "name": "VMDeploymentTypes",
        "value": "IaaS,PaaS"
      },
      {
        "name": "vCPUsAvailable",
        "value": "20"
      },
      {
        "name": "ACUs",
        "value": "230"
      },
      {
        "name": "vCPUsPerCore",
        "value": "2"
      },
      {
        "name": "CombinedTempDiskAndCachedIOPS",
        "value": "40000"
      },
      {
        "name": "CombinedTempDiskAndCachedReadBytesPerSecond",
        "value": "335544320"
      },
      {
        "name": "CombinedTempDiskAndCachedWriteBytesPerSecond",
        "value": "335544320"
      },
      {
        "name": "UncachedDiskIOPS",
        "value": "32000"
      },
      {
        "name": "UncachedDiskBytesPerSecond",
        "value": "503316480"
      },
      {
        "name": "EphemeralOSDiskSupported",
        "value": "True"
      },
      {
        "name": "EncryptionAtHostSupported",
        "value": "True"
      },
      {
        "name": "CapacityReservationSupported",
        "value": "True"
      },
      {
        "name": "ConfidentialComputingType",
        "value": "SNP"
      },
      {
        "name": "AcceleratedNetworkingEnabled",
        "value": "False"
      },
      {
        "name": "RdmaEnabled",
        "value": "False"
      },
      {
        "name": "MaxNetworkInterfaces",
        "value": "8"
      }
    ],
    "capacity": null,
    "costs": null,
    "family": "standardECADSv5Family",
    "kind": null,
    "locationInfo": [
      {
        "location": "eastus",
        "zoneDetails": [
          {
            "Name": [
              "2",
              "1"
            ],
            "capabilities": [
              {
                "name": "UltraSSDAvailable",
                "value": "True"
              }
            ],
            "name": null
          }
        ],
        "zones": [
          "2",
          "1"
        ]
      }
    ],
    "locations": [
      "eastus"
    ],
    "name": "Standard_EC20ads_v5",
    "resourceType": "virtualMachines",
    "restrictions": [],
    "size": "EC20ads_v5",
    "tier": "Standard"
  },
]
```